omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/PatientAllergyController2_0Test.java

```diff
@@ -0,0 +1,459 @@
+		Allergen allergen = new Allergen(AllergenType.DRUG, severity, "");
+		allergyToSave.setAllergen(allergen);
+		
+		// save allergy
+		String json = getMapper().writerWithView(JsonAllergyViews.class).writeValueAsString(allergyToSave);
+		System.out.println("<allergyToSave>" + json);
+		
+		// save allergy with reactions to get uuid for later saving of reactions
+		String returnedJson = handle(newPostRequest(getURI(), json)).getContentAsString();
+		System.out.println("<savedJson>" + returnedJson);
+		Allergy savedAllergy = getMapper().readValue(returnedJson, Allergy.class);
+
+		TestAllergyForEquality(allergyToSave, savedAllergy);
+
+		// reactions have to be set separately with allergy uuid
+		allergyToSave.setUuid(savedAllergy.getUuid());
+		allergyToSave.addReaction(new AllergyReaction(allergyToSave, severity, ""));
+		allergyToSave.addReaction(new AllergyReaction(allergyToSave, otherNonCoded, "test non coded"));
+		
+		// update allergy with reactions
+		json = getMapperAllergyUuid().writerWithView(JsonAllergyViews.class).writeValueAsString(allergyToSave);
+		System.out.println("<newAllergy>" + json);
+
+		// update allergy with reactions
+		returnedJson = handle(newPostRequest(getURI() + "/" + allergyToSave.getUuid(), json)).getContentAsString();
+		System.out.println("<newJson>" + returnedJson);
+		savedAllergy = getMapper().readValue(returnedJson, Allergy.class);
+
+		TestAllergyForEquality(allergyToSave, savedAllergy);
+		
+		allergies = Context.getPatientService().getAllergies(patient);
+		Assert.assertEquals(allergies.size(), 5);
+	}
+	
+	/**
+	 *  Save Allergy with Non Coded Allergen
+	 */
+	@Test
+	public void shouldSaveAllergyWithNonCodedAllergen() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Patient patient = allergy.getPatient();
+		Allergies allergies = Context.getPatientService().getAllergies(patient);
+		Assert.assertEquals(allergies.size(), 4);
+		
+		// create allergy with non coded allergen
+		Allergy allergyToSave = new Allergy();
+		allergyToSave.setComment("allergy comment");
+		Concept severity = new Concept();
+		Concept otherNonCoded = new Concept();
+		severity.setUuid("35d3346a-6769-4d52-823f-b4b234bac3e3");
+		otherNonCoded.setUuid("5622AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA");
+		allergyToSave.setSeverity(severity);
+		Allergen allergen = new Allergen(AllergenType.DRUG, otherNonCoded, "test non coded");
+		allergyToSave.setAllergen(allergen);
+		
+		// save allergy
+		String json = getMapper().writerWithView(JsonAllergyViews.class).writeValueAsString(allergyToSave);
+		System.out.println("<allergyToSave>" + json);
+		
+		// save allergy with reactions to get uuid for later saving of reactions
+		String returnedJson = handle(newPostRequest(getURI(), json)).getContentAsString();
+		System.out.println("<savedJson>" + returnedJson);
+		Allergy savedAllergy = getMapper().readValue(returnedJson, Allergy.class);
+
+		TestAllergyForEquality(allergyToSave, savedAllergy);

```

Comment:
Our convention is to have methods names that start with lower case https://wiki.openmrs.org/display/docs/Coding+Conventions